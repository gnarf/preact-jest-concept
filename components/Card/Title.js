import styled from 'styled-components';

const Title = styled.h2`
  font-size: 1rem;
  color: #3A3A3A;
`;

export default Title;