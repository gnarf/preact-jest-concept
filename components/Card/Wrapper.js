import styled from 'styled-components';

const Wrapper = styled.div`
  width: 100%;
  border: 1px solid rgba(0,0,0,0.2);
`;

export default Wrapper;