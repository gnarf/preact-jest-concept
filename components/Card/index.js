import React from 'react';

import Wrapper from './Wrapper';
import Title from './Title';

const Card = ({title}) => (
  <Wrapper>
    <Title>{title}</Title>
  </Wrapper>
);

export default Card;