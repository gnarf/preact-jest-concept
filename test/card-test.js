import React from 'react';
import 'jest-styled-components';
import { shallow } from 'preact-render-spy';

import Card from '../components/Card';

test('Renders the card', () => {
  const context = shallow(<Card title="Hello people" />);

  context[0].$$typeof = Symbol.for('react.test.json')

  expect(context[0]).toMatchSnapshot();
});