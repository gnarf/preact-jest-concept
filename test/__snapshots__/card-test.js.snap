// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the card 1`] = `<cl title="Hello people"></cl>`;
